src/adam_impact_study/conversions.py

```diff
@@ -162,75 +119,144 @@ def sorcha_output_to_od_observations(sorcha_observations_df):
     -------
     fo_file_name : str
         Path to the generated Find_Orb input file.
     """
     with open(fo_file_name, "w") as w:
-        w.write("trkSub|stn|obsTime|ra|dec|rmsRA|rmsDec|mag|rmsMag|band\n")
-        for index, row in sorcha_df.iterrows():
-            time_tai = Timestamp.from_mjd([row["fieldMJD_TAI"]], scale="tai")
-            time_utc = time_tai.rescale("utc")
+        w.write("trkSub|stn|obsTime|ra|dec|rmsRA|rmsDec\n")  # |mag|rmsMag|band
+        for obs in od_observations:
+            sigmas = obs.coordinates.covariance.sigmas
+            time_utc = obs.coordinates.time
             time = time_utc.to_astropy()
             w.write(
-                f"{row['ObjID']}|X05|{time.isot[0]}|{row['RA_deg']}|"
-                f"{row['Dec_deg']}|"
-                f"{float(row['astrometricSigma_deg'])*3600}|"
-                f"{float(row['astrometricSigma_deg'])*3600}|"
-                f"{row['trailedSourceMag']}|"
-                f"{row['trailedSourceMagSigma']}|"
-                f"{row['optFilter']}\n"
+                f"{obs.object_id}|X05|{time.isot[0]}|{obs.coordinates.lon[0]}|"
+                f"{obs.coordinates.lat[0]}|"
+                f"{format(sigmas[0][1]*3600, '.5f')}|"
+                f"{format(sigmas[0][2]*3600, '.5f')}|"
+                f"{obs.photometry.mag}|"
+                f"{obs.photometry.mag_sigma}|"
+                f"{obs.photometry.filter}\n"
             )
     return fo_file_name
 
 
-def fo_to_adam_orbit_cov(elements_dict, covar_dict):
+def read_fo_output(fo_output_dir):
     """
-    Convert Find_Orb output to ADAM Orbit objects, including covariance.
+    Read the find_orb output files from the specified directory into dictionaries
+    containing orbital elements and covariances.
 
     Parameters
     ----------
+    fo_output_dir : str
+        Directory path where find_orb output files (e.g., total.json and covar.json) are located.
+
+    Returns
+    -------
     elements_dict : dict
-        Dictionary containing orbital elements for each object, keyed by Object ID.
+        Dictionary containing orbital elements for each object.
     covar_dict : dict
-        Dictionary containing covariance matrices and state vectors for each object.
+        Dictionary containing covariance matrices for each object.
+    """
+    covar_dict = read_fo_covariance(f"{fo_output_dir}/covar.json")
+    elements_dict = read_fo_orbits(f"{fo_output_dir}/total.json")
+    return elements_dict, covar_dict
+
+
+def read_fo_covariance(covar_file):
+    """
+    Read the find_orb covariance JSON file into a dictionary.
+
+    Parameters
+    ----------
+    covar_file : str
+        Path to the find_orb covariance JSON file (covar.json).
+
+    Returns
+    -------
+    covar_json : dict
+        Dictionary containing the covariance data from the JSON file.
+    """
+    with open(covar_file, "r") as f:
+        covar_json = json.load(f)
+    return covar_json
+
+
+def read_fo_orbits(input_file):
```

Comment:
It might be good to add to the documentation here a note about what the structure is of the loaded json and what the various components represent, to the best of our knowledge.